lib/please_cli/please_cli/base_image.py

```diff
@@ -66,15 +70,44 @@
     default=[],
     help='Public key for nix cache',
     )
+@click.option(
+    '--dont-push',
```

Comment:
Nit: why not use `--push` that defaults to `False` ? That would make line 168 as `if push:`  instead of `if not dont_push:`

Reply:
i actually started with `--push` since it makes more sense in code, but from a command line perspective you want to push by default and have an option to not push.